README.md

```diff
@@ -1,273 +1,440 @@
+4. **Define Benchmark Method**: Next, create methods that you would like to be benchmarked within this class and annotate them with `@Benchmark`.
 
-Define a new compilation in whichever target you'd like (e.g. `jvm`, `js`, etc):
-```groovy
-kotlin {
-    jvm {
-        compilations.create('benchmark') { associateWith(compilations.main) }
+    ```kotlin
+    @Benchmark
+    fun benchmarkMethod(): Int {
+        return list.sum()
     }
-}
-```
+    ```
+
+Your final benchmark class will look something like this:
+
+    @State(Scope.Benchmark)
+    class MyBenchmark {
+
+        private val size = 10
+
+        private val list = ArrayList()
+
+        @Setup
+        fun prepare() {
+            for (i in 0 until size) {
+                list.add(i)
+            }
+        }
+
+        @Benchmark
+        fun benchmarkMethod(): Int {
+            return list.sum()
+        }
+
+        @TearDown
+        fun cleanup() {
+            list.clear()
+        }
+    }
+
+Note: Benchmark classes located in the common source set will be run in all platforms, while those located in a platform-specific source set will be run only in the corresponding platform.
+
+See [writing benchmarks](docs/writing-benchmarks.md) for a complete guide for writing benchmarks.
 
-Register it by its source set name (`jvmBenchmark` is the name for the `benchmark` compilation for `jvm` target):
+### Running Benchmarks
 
-```groovy
+To run your benchmarks in all registered platforms, run `benchmark` Gradle task in your project.
+To run only on a specific platform, run `<target-name>Benchmark`, e.g., `jvmBenchmark`.
+
+For more details about the tasks created by the kotlinx-benchmark plugin, refer to [this guide](docs/tasks-overview.md).
+
+### Benchmark Configuration Profiles
+
+The kotlinx-benchmark library provides the ability to create multiple configuration profiles. The `main` configuration is already created by the Toolkit.
+Additional profiles can be created as needed in the `configurations` section of the `benchmark` block:
+
+```kotlin
+// build.gradle.kts
 benchmark {
-    targets {
-        register("jvmBenchmark")    
+    configurations {
+        named("main") {
+            warmups = 20
+            iterations = 10
+            iterationTime = 3
+            iterationTimeUnit = "s"
+        }
+        register("smoke") {
+            include("<pattern of fully qualified name>")
+            warmups = 5
+            iterations = 3
+            iterationTime = 500
+            iterationTimeUnit = "ms"
+        }
     }
 }
-```
+```  
+
+Refer to our [comprehensive guide](docs/configuration-options.md) to learn about configuration options and how they affect benchmark execution.
+
+### Separate source sets for benchmarks
+
+Often you want to have benchmarks in the same project, but separated from main code, much like tests.
+Refer to our [detailed documentation](docs/separate-source-sets.md) on configuring your project to add a separate source set for benchmarks.
```

Comment:
Not sure if we can merge it without either adding all new docs from `docs`, or leaving the old description.

Reply:
This documentation is added in a separate PR: https://github.com/Kotlin/kotlinx-benchmark/pull/175